x-pack/plugins/actions/server/index.ts

```diff
@@ -56,8 +56,33 @@ export const plugin = (initContext: PluginInitializerContext) => new ActionsPlug
 
 export const config: PluginConfigDescriptor<ActionsConfig> = {
   schema: configSchema,
+  exposeToUsage: {
+    whitelistedHosts: true,
+    rejectUnauthorized: true,
+    proxyRejectUnauthorizedCertificates: true,
+  },
   deprecations: ({ renameFromRoot, unused }) => [
-    renameFromRoot('xpack.actions.whitelistedHosts', 'xpack.actions.allowedHosts'),
+    // Use a custom copy function here so we can perserve the telemetry provided for the deprecated config
+    // See https://github.com/elastic/kibana/issues/112585#issuecomment-923715363
+    (settings, fromPath, addDeprecation) => {
+      const fullOldPath = 'xpack.actions.whitelistedHosts';
+      const fullNewPath = 'xpack.actions.allowedHosts';
+      const actions = get(settings, fromPath);
+      const whitelistedHosts = actions?.whitelistedHosts;
+      if (whitelistedHosts) {
```

Comment:
I tested using both the new and then the old settings, and saw the deprecation notice as it was intended.  I also happened to use both in my config, and was slightly surprised that it didn't complain - the values on the two keys were different.  Feels like this is something we could test for and warn about - I'd think we'd want to ignore the old in this case.

Reply:
Yea makes sense. I had to copy the implementation for our custom `copy` deprecation method and I think I might have missed that. Good catch!